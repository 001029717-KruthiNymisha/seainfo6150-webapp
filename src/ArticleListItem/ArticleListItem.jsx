import React, { useState } from 'react';
import HTMLText from "../HTMLText/HTMLText";
import styles from "./ArticleListItem.module.css";
import ArticleImage from "../ArticleImage/ArticleImage.jsx";

import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";

const ArticleListItem = (prop) => {
 let buttonText = 'Show more';
    const [toggleData, setToggleText] = useState(false);
  return (
          <li  className={styles.container}>
          <article className={styles.article}>
           <ArticleImage url = {prop.article.image._url} title={prop.article.title}></ArticleImage>
            <h1 className={styles.title}>{prop.article.title}</h1>
{toggleData &&
                      <div>

                        <p className={styles.shortText}>{prop.article.shortText} </p>
                        <time className={styles.displyDate} datetime={prop.article.displyDate}>{prop.article.displayDate} </time>
                        <br/><br/>
                      </div>
                    }
           <ArticleTextToggleButton
                     buttonText = {toggleData?'Show less':'Show more'}
                     onClick={()=>{
                       if(toggleData){
                         setToggleText(false);
                       }else{
                         setToggleText(true);
                       }
                   }}>
           </ArticleTextToggleButton>
           </article>
          </li>

  );
};

export default ArticleListItem;
