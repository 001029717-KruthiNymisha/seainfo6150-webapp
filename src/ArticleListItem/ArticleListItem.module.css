.container {
    background: white;
    background-image: linear-gradient(to right, rgb(179 37 37), white 10%);
    border: 1px solid rgb(179 37 37);
    border-radius: 5px;
      margin-top: 4px;
      margin-bottom: 4px;
      margin-right: 10px;
      margin-left: -25px;
}
@media (min-width: 300px) {
  .container {
   background: white;
   background-image: linear-gradient(to right,#D2691E, white 10%);
  }
}

@media (min-width: 650px) {
  .container {
     background: white;
     background-image: linear-gradient(to right,#DC143C, white 10%);
  }
}

@media (min-width: 1000px) {
  .container {
     background: white;
     background-image: linear-gradient(to right,rgb(179 37 37), white 10%);
     }
}

.article {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-between;
  height: 100%;
}

.title {
    text-decoration: underline;
    color: rgb(179 37 37);
    font-size: 18px;
    padding-left: 10px;
    padding-bottom: 10px;
}
@media (min-width: 300px) {
  .title {
      color: #7e3f12;
      font-size: 18px;
  }
}

@media (min-width: 650px) {
  .title {
        color: #8D0D26;
        font-size: 18px;
    }
}

@media (min-width: 1000px) {
  .title {
        color: rgb(179 37 37);
        font-size: 18px;
    }
}

.shortText {
    color: black;
    font-size: 14px;
    margin-left: 10px;
}

.displyDate {
    font-weight: bold;
    margin-left: 10px;
    margin-bottom: 30px;
}

