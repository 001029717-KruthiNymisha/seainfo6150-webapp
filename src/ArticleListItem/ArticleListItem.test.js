import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";
describe("ArticleListItemTests", () => {
    const article = {
        slug: '181122102925-statue-of-liberty-crown',
        title: "The Statue of Liberty's torch heads to new museum",
        shortText:
            "The Statue of Liberty's original torch is getting a new home",
        author: 'Kate Farley',
        displayDate: 'November 22nd 2018, 7:12 am',
                 image: {
                          _url: "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg",
                        },
    };
    it("renders correctly", () => {
        const { container } = render(<ArticleListItem article={article} />);
        expect(container).toMatchSnapshot();
    });
});