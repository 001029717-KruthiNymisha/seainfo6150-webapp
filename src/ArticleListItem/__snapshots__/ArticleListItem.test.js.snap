// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItemTests renders correctly 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div>
        <img
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <h1
          class="title"
          src="The Statue of Liberty's torch heads to new museum"
        />
      </div>
      <h1
        class="title"
      >
        The Statue of Liberty's torch heads to new museum
      </h1>
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
